src/components/Home.tsx

```diff
@@ -0,0 +1,19 @@
+import { useSelector } from 'react-redux';
+
+interface User {
+    username: string;
+}
+
+interface UserState {
+    user: User | null;
+}
+
+export default function Home() {
+    const user = useSelector((state: { user: UserState }) => state.user.user);
```

Comment:
1. You might want to consider creating a selector function (`getUsername(state)`, used as `const username = useSelector(getUsername);` or `const {username} = useSelector(getUser)` or similar)
2. The structure is a little awkward - `state.user.user`, it might be better to avoid the duplication